File: behave/features/steps/calculate.steps.py
from behave import given, then, when


@given("I have the number {initialValue}")
def given_number(context, initialValue: str):
    context.number1 = int(initialValue)


@when("I increase it by {value}")
def when_increase(context, value: str):
    context.result = context.number1 + int(value)


@when("I subtract {value} from it")
def when_subtract(context, value: str):
    context.result = context.number1 - int(value)


@then("the result should return {expected}")
def then_return(context, expected: str):
    assert context.result == int(expected)

File: cucumber/features/steps/calculate.steps.js
const {
  Given,
  When,
  Then,
  After,
} = require("@cucumber/cucumber");
const assert = require("assert");

let number1, number2, result;

Given("I have the number {int}", function (num) {
  number1 = num;
});

Given(
  "I have another number {int}",
  function (num) {
    number2 = num;
  },
);

When("I increase it by {int}", function (number2) {
  result = number1 + number2;
});

When(
  "I subtract {int} from it",
  function (number2) {
    result = number1 - number2;
  },
);

Then(
  "the result should return {int}",
  function (expectedResult) {
    assert.strictEqual(result, expectedResult);
  },
);

After(function () {
  this.attach('{"name": "some JSON"}', {
    mediaType: "application/json",
  });
});
